### PR TITLE
New hello world

### PR DESCRIPTION
**Pull request by Rich-Harris**
Following up @rafaelcamaram's work in #1007 with a few changes:

* [x] make it possible to do `pnpm dev` inside the template, for easier developing
* [x] fix accessibility of welcome image
* [x] use spring physics for the counter (transitions don't really work here)
* [x] ensure that hero image has dimensions set to prevent layout popping
* [x] use localStorage mechanism for dark mode, use CSS to swap welcome image
* [x] update welcome images (@vedam — we're currently using pngs ripped straight out of the [figma document](https://www.figma.com/file/atJYebOMeqoS6MWzxjOfE4/svelte-template), but the light/dark mode welcome images don't line up perfectly, and it looks like the light mode one has some scaling artifacts. Do you happen to have originals handy?
* [x] populate about page with something
* [x] populate blog section with something

It would also be cool if this template was the demo app referred to in #639. To that end we _may_ want to test/showcase various features:

* dynamic endpoints
* prerendering
* non-hydrated pages
* etc

### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts

### CODE REVIEW

**Comment by benmccann** [comment]
Looks like you need to run prettier on `create-svelte`

File: packages/create-svelte/templates/skeleton/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.DS_Store
+node_modules
+/.svelte
+/build
+/functions
```
Comment (benmccann) [inline packages/create-svelte/templates/skeleton/.gitignore:5]: might be nice to add a new line at the end of the file

File: packages/create-svelte/templates/skeleton/.meta.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+	"description": "Demo project"
```
Comment (benmccann) [inline packages/create-svelte/templates/skeleton/.meta.json:2]: I can't find anywhere online that talks about `.meta.json`. Is this a VS Code thing?

File: packages/create-svelte/templates/skeleton/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+npm run dev -- --open
+```
+
+## Building
+
+Svelte apps are built with _adapters_, which optimise your project for deployment to different environments.
+
+By default, `npm run build` will generate a Node app that you can run with `node build`. To use a different adapter, add it to the `devDependencies` in `package.json` making sure to specify the version as `next` and update your `svelte.config.cjs` to [specify your chosen adapter](https://kit.svelte.dev/docs#configuration-adapter). The following official adapters are available:
+
+- [@sveltejs/adapter-node](https://github.com/sveltejs/kit/tree/master/packages/adapter-node)
```
Comment (benmccann) [inline packages/create-svelte/templates/skeleton/README.md:36]: should we add the cloudflare adapter? Or maybe just link to https://kit.svelte.dev/docs#adapters and add it there?

File: packages/create-svelte/templates/default/src/lib/Counter/minus.svg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="2" viewBox="0 0 16 2" fill="black" stroke="black" xmlns="http://www.w3.org/2000/svg">
+	<path d="M1.25 1H15.25" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" />
+</svg>
```
Comment (benmccann) [inline packages/create-svelte/templates/default/src/lib/Counter/minus.svg:3]: should we add a newline to the end of this file and `plus.svg`?

**Comment by benmccann** [comment]
lgtm. thanks @rafaelcamaram for blazing the trail here!

**Comment by vedam** [comment]
> * update welcome images – @vedam .... Do you happen to have originals handy?
I'll look for them asap.

File: packages/create-svelte/templates/default/src/lib/Counter/minus.svg (hunk 4 above)
Comment (Rich-Harris) [reply packages/create-svelte/templates/default/src/lib/Counter/minus.svg]: ended up removing them

File: packages/create-svelte/templates/skeleton/README.md (hunk 3 above)
Comment (Rich-Harris) [reply packages/create-svelte/templates/skeleton/README.md]: added it to the README

File: packages/create-svelte/templates/skeleton/.meta.json (hunk 2 above)
Comment (Rich-Harris) [reply packages/create-svelte/templates/skeleton/.meta.json:2]: it's so that we can add a description of the templates when prompting

Comment (benmccann) [reply packages/create-svelte/templates/skeleton/.meta.json:2]: ok, I missed that. it'd probably be nice if this wasn't added to the user's project. I didn't look at the code, so perhaps that's already happening

**Comment by benmccann** [comment]
Perhaps a bit off-topic, but the thing that's always bugged me about create-svelte is that it dumps the files in your current directory. I'd like it if it asked you for your project name and created a new directory and put the files there

**Comment by Rich-Harris** [comment]
It does - `npm init svelte@next my-app`

File: packages/create-svelte/cli/index.js [hunk 5]
```diff
@@ -37,6 +37,15 @@ async function main() {
 	}
 
 	const options = /** @type {import('./types').Options} */ (await prompts([
+		{
+			type: 'select',
+			name: 'template',
+			message: 'Which app Svelte template?',
```
Comment (dummdidumm) [inline packages/create-svelte/cli/index.js:43]:
```suggestion
			message: 'Which Svelte app template?',
```

File: packages/create-svelte/templates/default/src/lib/Header/ThemeToggler.svelte [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+	}
+
+	svg {
+		width: 2em;
+		height: 2em;
+	}
+
+	path {
+		fill: black;
+		/* transition: opacity 0.6s; */
```
Comment (AnatoleLucet) [inline packages/create-svelte/templates/default/src/lib/Header/ThemeToggler.svelte:67]:
```suggestion
```

**Comment by Rich-Harris** [comment]
One thing that occurred to me — the theme toggler makes it impossible to use `hydrate = false`. We need to decide which is more important to showcase

**Comment by dummdidumm** [comment]
I feel that `hydrate = false` is more important, a theme toggler is a nice eye catcher but not unique functionality.

**Comment by Rich-Harris** [comment]
Makes sense. So, next question: dark theme or light theme? I'm quite partial to the dark theme myself

**Comment by vedam** [comment]
Personal preference IMO. I personally find dark themes harder to read with loads of Text.
Also we should consider that the Svelte- and Kit-Site are light-themed. (The corporate-design-voice in me)

File: packages/create-svelte/templates/skeleton/.meta.json (hunk 2 above)
Comment (Rich-Harris) [reply packages/create-svelte/templates/skeleton/.meta.json:2]: it is not added to the user's project

**Comment by rafaelcamaram** [comment]
I would say that, if we need to choose, would be better to keep a light theme -- since there are some articles saying that dark-theme lead us to a different set of accessibility issues for people with some sort of astigmatism (something common for developers)

**Comment by vedam** [comment]
> * update welcome images @vedam

So if I got it right, it seems we go with the light theme?
Means we also need only the light welcome-rendering.
I found some render-glitches and fixed them in the final one.
It's still a bit larger than needed, but I think it's always better to downsize than the other way.
welcome-4

**Comment by Rich-Harris** [comment]
Added that artwork. It's looking good! Thanks @vedam. I compressed it with https://squoosh.app and am using `<picture>` to use webp with png fallback; it _would_ be pretty awesome if the framework had a turnkey way to import multiple renditions/formats of an image asset automatically, but that's a conversation for another day.

https://user-images.githubusercontent.com/1162160/114754933-19f98380-9d27-11eb-8761-87c14d565579.mov

**Comment by vedam** [comment]
Great, looks awesome.
Love the counter-anim. 🌶

> it _would_ be pretty awesome if the framework had a turnkey way to import multiple renditions/formats of an image asset automatically

Yes, that's rocket science. ;-)
I personally use [lazysizes](https://github.com/aFarkas/lazysizes) in combo with Cloudinary-CDN (generous free account). 
Place one big png there and it delivers every needed optimized Image/size/format on-the-fly. Be it mobile or big desktop. Squeezes my pictures to the max. And works with movies too.
`</end of promo-session>`

**Comment by Rich-Harris** [comment]
Rather than having a `/blog` page, which is a bit yawn-worthy, I thought it might be good to have a `/todos` page that allows us to demonstrate a few things:

* reading/writing cookies
* loading data from an API
* handling form submissions, with JS enabled or disabled
* transitions (and _maybe_ animations)

For now I'll probably hack something together with supabase or something, but it would be cool if there was an endpoint on api.svelte.dev reserved for this so that people don't need to configure backends for their own apps. The requirements would be very simple

**Comment by benmccann** [comment]
I've been trying to help a user get Supabase working with Vite the past couple of days. It's been a bit of a nightmare. We've fixed multiple issues such as Supabase sourcemaps being broken, but there are still two Vite bugs preventing it from running. Getting packages to work with Vite has been a huge pain point in general. In this case one of the issues is that it's using the browser field in package.json to resolve imports during ssr which I have a PR out for. Once that's fixed there's another issue that hasn't been filed yet and I haven't looked into and might be past my level of Vite knowledge. I don't know what the issue is but @GrygrFlzr suggested he wonders whether Vite might choke on packages whose entry point is a single line.

It might be a lot to make people to have to setup a backend to run Hello World. At most I would think maybe we could make sure the app still works if the connection fails and then give instructions to download a Docker container or something

**Comment by Rich-Harris** [comment]
Wouldn't be using the Supabase client, would just be making requests to its API from an endpoint.



> It might be a lot to make people to have to setup a backend to run Hello World

...which is why I said this!

> For now I'll probably hack something together with supabase or something, but it would be cool if there was an endpoint on api.svelte.dev reserved for this so that people don't need to configure backends for their own apps. The requirements would be very simple

**Comment by Rich-Harris** [comment]
Made a bit more progress:

https://user-images.githubusercontent.com/1162160/114972840-40194380-9e4d-11eb-8397-0c1aceb77d1a.mov

Open to design feedback as ever.

It demonstrates progressively enhanced forms, i.e. it works without JS. (The `enhance` action should probably live somewhere else, but I'm not sure where.) I haven't got round to setting a cookie that can be used as a user ID, will do that tomorrow.

Right now the data lives in a supabase Postgres database that I have the keys to...

image

...but if we could get this data served from api.svelte.dev, that would be ideal, so that it works for everyone who does `npm init svelte`. 

Thoughts welcome

**Comment by Rich-Harris** [comment]
Oh, and it's using the [method override workaround](https://github.com/sveltejs/kit/issues/1046) — would like to get this into the framework itself before 1.0

**Comment by vedam** [comment]
Design feedback
- Headlines on "About" and "TODOs" should have the same size than "to your new..." on "Home". Centering them would visually connect the page with the menu.
- Increasing line-height on "About"-Content would make the massive text-block a bit more lightweight (see svelte-site)
- `npx init svelte@next` should have a light background IMO
Bildschirmfoto 2021-04-16 um 07 19 23
Bildschirmfoto 2021-04-16 um 07 19 54

One more thing...
In our svelte-tutorials we have that outstanding [deferred-example](https://svelte.dev/tutorial/deferred-transitions) which differentiated a svelte-todo-list from all other todo-lists I've seen so far (and maybe will ever see in the future). It made click for me why "transitions included" is such a good idea.
I don't know if that would be too heavy for a starter-template.
Just a thought. (The best 'hello world' you've ever seen)  😁

File: packages/create-svelte/shared/README.md [hunk 7]
```diff
@@ -35,6 +35,7 @@ By default, `npm run build` will generate a Node app that you can run with `node
 
 - [@sveltejs/adapter-node](https://github.com/sveltejs/kit/tree/master/packages/adapter-node)
 - [@sveltejs/adapter-static](https://github.com/sveltejs/kit/tree/master/packages/adapter-static)
+- [@sveltejs/adapter-cloudflare-workers](https://github.com/sveltejs/kit/tree/master/packages/adapter-cloudflare-workers)
```
Comment (benmccann) [inline packages/create-svelte/shared/README.md:38]: Should we add this to https://github.com/sveltejs/kit/blob/master/documentation/docs/10-adapters.md instead or in addition? We could point this readme to that file

Comment (Rich-Harris) [reply packages/create-svelte/shared/README.md]: yep, have updated the README and opened https://github.com/sveltejs/kit/pull/1056

**Comment by Rich-Harris** [comment]
Thanks @vedam — have made those changes:

image

image

I did wonder about the deferred transition example — I _was_ worried that it might be a bit too much for an introductory page, since it's quite advanced stuff that isn't totally obvious on first glance.

We're tracking the API stuff on https://github.com/sveltejs/api.svelte.dev/issues/2. Once that's done, next step will be to figure out a way to deploy this demo app with the various adapters — it can take the place of https://github.com/sveltejs/kit/tree/master/examples/svelte-kit-demo (or this app could go there, and the `create-svelte` build process could read from that directory instead of its own `templates/default`).

File: packages/create-svelte/templates/default/src/lib/db.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
+import fetch from 'node-fetch';
```
Comment (dummdidumm) [inline packages/create-svelte/templates/default/src/lib/db.ts:1]: I'm still a bit unsure about hitting a real api. Couldn't we do a mock here instead and write to local storage? If everyone else uses the db interface we can showcase these things just the same.

**Comment by dummdidumm** [comment]
A general thought: should the the app be enhanced with a couple of comments explaining the various features showcased here with a link to the docs? Like "the load function can be used to fetch data. See -link- for more info"

File: packages/create-svelte/templates/default/src/lib/db.ts (hunk 8 above)
Comment (Rich-Harris) [reply packages/create-svelte/templates/default/src/lib/db.ts:1]: You can't make an app that works without JS if you rely on localStorage

**Comment by Rich-Harris** [comment]
Yep, added some docs, and we can certainly add more

File: packages/create-svelte/package.json [hunk 9]
```diff
@@ -2,24 +2,23 @@
...
-		"prompts": "^2.4.0",
-		"rollup": "^2.41.1",
 		"sucrase": "^3.18.1",
 		"svelte-preprocess": "^4.7.0",
 		"tiny-glob": "^0.2.8"
 	},
 	"scripts": {
-		"dev": "rollup -cw",
-		"build": "node scripts/build-templates && rollup -c",
+		"build": "node scripts/build-templates",
```
Comment (dummdidumm) [inline packages/create-svelte/package.json:21]: Now that the code is no longer completely contained inside `bin.js`, you need to update `files` below accordingly - `utils.js` needs to be added there.

**Comment by Rich-Harris** [comment]
API is rigged up, so I'll merge this in even though there's doubtless room for improvement. Next stop after this is released will be to get it deployed with the various different adapters

**Pull request merged**